### ISSUE TITLE
Position indicator and line behavior inconsistent between Song-Editor, Piano-Roll, Beat/Bassline and Automation Editor

### ISSUE THREAD

**Issue by simonvanderveldt** [LMMS/lmms#3052]
All three of the Song-Editor, Piano-Roll and Automation Editor show the the position arrow/indicator in the timeline.
For the Song-Editor and the Piano-Roll the position arrow/indicator moves when you start playing the song from within them. For the Automation Editor it doesn't, it just stays at the leftmost position.

Also, only in the Song-Editor there is a position line that moves along the track when playing to show the current position in the track, this position line doesn't exist in the Piano-Roll and the Automation Editor. IMHO having the position line makes it a lot clearer what's currently being played, so I'd like to propose to add it to the Piano-Roll and Automation Editor as well.

**Comment by musikBear** [comment]
#857 doublet

imo a vertical line in piano-roll is a bit of an overhead, but if its possible to implement without loss of efficiency, then np..

**Comment by simonvanderveldt** [comment]
@musikBear Thanks for the reference, I did a search but couldn't find anything related to it.

What do you mean by overhead?

**Comment by tresf** [comment]
> #857 doublet

Agreed.  Closed #857 in favor of this new one. 👍 

> What do you mean by overhead?

I assume he's talking CPU usage, but he's not a coder, so @musikBear let's leave that type of speculation for the coders. 👍

**Comment by miketurn** [comment]
I agree, having a position indicator in these windows would be very helpful.

The piano roll contains just the down pointing arrow "position indicator" but not the vertical white line.
I would imagine this was not added due to maybe conflicting a little bit with the grid in the background, but it would still be nice to have one shown, maybe making it a different color, like a bright LMMS green or something so it stands out above the grid, if the white does not work.

Also I see it was added to the title, but not really mentioned here, but it might also be nice to see a "position indicator" added to the B+B Editor as well because there is no visible indicator there.

As with any new features, I would say having a show/hide ability for the "position indicator" would probably be a good idea for those who maybe would find it distracting.

Thank You for considering these ideas.

**Comment by tresf** [comment]
As part of a pruning effort, this `enhancement` request is archived into a dedicated "Better Workflow" checklist here #4877.

**Comment by FyiurAmron** [comment]
@tresf (or anyone else): in 1.3.0-alpha in Pattern Editor (i.e. "new" renamed Beat/Bassline Editor), I can't find any kind of position indicator nor timeline. FWIW, I'm not seeing it in 1.2.2 (stable) B/B Editor neither.
Am I missing something obvious, or it's some kind of regression or bug (as all the ticket related to this functionality it are now closed)?

**Comment by Rossmaxx** [comment]
This issue is now a part of #4877 meta issue. We are tracking this there.

**Comment by FyiurAmron** [comment]
@Rossmaxx do you maybe know if is anyone actively working on it or did the initial research what needs to be done here? I've just been able to get master to compile on my box and have some spare brainpower and can take a look what's going on in B/B, at least some preliminary scouting, if no-one did that before - I could really use this feature, so I'm more than interested in shipping it even if I'd have to do the coding myself : ) (Dunno if this would be a "Good first issue for new developers", but still)

**Comment by Rossmaxx** [comment]
Go ahead. Contributions are welcome. We are all busy with other aspects of LMMS. For instance, I'm trying to modenize the build tooling with modernizing cmake and porting it to qt 6. I'm not really familiar with the gui. You can proceed with your work and open a PR.

**Comment by tresf** [comment]
> I can't find any kind of position indicator nor timeline [in Pattern editor]

@miketurn actually pointed this oddity out, quoting:

> Also I see it was added to the title, but not really mentioned here, but it might also be nice to see a "position indicator" added to the B+B Editor as well because there is no visible indicator there.

With regards to who's working on this, it's as good as yours.  The puning effort was an attempt to clean stuff up that -- at that time -- seemed to be tackled by larger UI overhauls.  That never happened, so feel free to request any bugs reopened that you want to work on. 🍻
